### ISSUE TITLE
CentOS-Base.repo file is missing

### ISSUE THREAD

**Issue by abraverm** [openshift/jenkins#1093]
Building `jenkins/2/Dockerfile.localdev` fails due to a file that was removed:
https://raw.githubusercontent.com/cloudrouter/centos-repo/master/CentOS-Base.repo

The error:
```
--> FROM openshift/origin-release:golang-1.12 as go-init-builder
--> ENV "GIT_SSL_NO_VERIFY"="true" "TENANT_NAME"="my-jenkins" "JENKINS_SERVICE_NAME"="my-jenkins-jenkins"
--> WORKDIR /go/src/github.com/openshift/jenkins
--> COPY . .
--> WORKDIR /go/src/github.com/openshift/jenkins/go-init
--> RUN go build . && cp go-init /usr/bin
--> FROM quay.io/openshift/origin-cli as 1
--> ENV "GIT_SSL_NO_VERIFY"="true" "TENANT_NAME"="my-jenkins" "JENKINS_SERVICE_NAME"="my-jenkins-jenkins"
--> MAINTAINER Akram Ben Aissi <abenaiss@redhat.com>
--> COPY --from=go-init-builder /usr/bin/go-init /usr/bin/go-init
--> ENV JENKINS_VERSION=2 HOME=/var/lib/jenkins JENKINS_HOME=/var/lib/jenkins JENKINS_UC=https://updates.jenkins.io OPENSHIFT_JENKINS_IMAGE_VERSION=4.0 LANG=en_US.UTF-8 LC_ALL=en_US.UTF-8 INSTALL_JENKINS_VIA_RPMS=false
--> LABEL k8s.io.description="Jenkins is a continuous integration server" k8s.io.display-name="Jenkins 2" openshift.io.expose-services="8080:http" openshift.io.tags="jenkins,jenkins2,ci" io.jenkins.version="2.190.2" io.openshift.s2i.scripts-url=image:///usr/libexec/s2i
--> EXPOSE 8080 50000
--> RUN curl https://raw.githubusercontent.com/cloudrouter/centos-repo/master/CentOS-Base.repo -o /etc/yum.repos.d/CentOS-Base.repo &&     curl http://mirror.centos.org/centos-7/7/os/x86_64/RPM-GPG-KEY-CentOS-7 -o /etc/pki/rpm-gpg/RPM-GPG-KEY-CentOS-7 &&     INSTALL_PKGS="dejavu-sans-fonts rsync gettext git tar zip unzip openssl bzip2 java-11-openjdk java-11-openjdk-devel jq" &&     DISABLES="--disablerepo=rhel-server-extras --disablerepo=rhel-server --disablerepo=rhel-fast-datapath --disablerepo=rhel-server-optional --disablerepo=rhel-server-ose --disablerepo=rhel-server-rhscl" &&     yum $DISABLES -y --setopt=tsflags=nodocs --disableplugin=subscription-manager install epel-release &&     yum $DISABLES -y --setopt=tsflags=nodocs --disableplugin=subscription-manager install $INSTALL_PKGS &&     rpm -V  $INSTALL_PKGS &&     yum clean all  &&     localedef -f UTF-8 -i en_US en_US.UTF-8
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100    14  100    14    0     0     27      0 --:--:-- --:--:-- --:--:--    27
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  1690  100  1690    0     0  21932      0 --:--:-- --:--:-- --:--:-- 22236
Loaded plugins: ovl, product-id, search-disabled-repos
File contains no section headers.
file: file:///etc/yum.repos.d/CentOS-Base.repo, line: 1
'404: Not Found'
```

**Comment by akram** [comment]
thanks for reporting this. working on a fix now

**Comment by abraverm** [comment]
I hope the original file will return as I'm using an older version of the repo (around 2.190.2 bump commit)

**Comment by abraverm** [comment]
is there a way to restore the original file?

**Comment by akram** [comment]
@abraverm I am afraid it is not possible. I don't have the historical reason on why we used this repository, but for sure we (Jenkins on OpenShift / Red Hat team) are not the owners of the repo there, and this dependency was there for a long time ago.


I have put a copy of the file in our current repository so, we shouldn't face the issue anymore.
Please, grab it from our main branch and tell me if it works. I will close the issue as it is fixed by #1094 

Let me know if I should reopen it. We will not do backport unless it is required for solving a security issue in the future.

**Comment by michaelsauter** [comment]
@akram I just came across this issue. While this repository has been updated recently, and the RHEL based image was also pushed recently (28 days ago), the last push on Docker hub for `openshift/jenkins-2-centos7:v3.11` was 4 months ago. It seems that the image push doesn't work - as can be seen in https://ci.openshift.redhat.com/jenkins/view/All/job/push_jenkins_images/335/console, which shows the exact same error @abraverm posted above. As a result, the latest available image has Jenkins in version 2.204 and outdated plugins, which leads to an issue we are facing right now (https://github.com/opendevstack/ods-jenkins-shared-library/issues/413). Can we get the fixes you made in #1094 to be applied so that we have a centos image in Docker Hub that is up-to-date and aligned with the RHEL one?
